### PR TITLE
feat: scrollTo allows ElementHandle arg

### PR DESCRIPTION
**Pull request by bvandrc**

### CODE REVIEW

**Comment by Niek** [comment]
@bvandrc can you please check review comments?

File: src/spoof.ts [hunk 1]
```diff
@@ -159,8 +159,6 @@ export interface MoveToOptions extends PathOptions, Pick<MoveOptions, 'moveDelay
   readonly moveDelay?: number
 }
 
-export type ScrollToDestination = Partial<Vector> | 'top' | 'bottom' | 'left' | 'right'
-
 export type MouseButtonOptions = Pick<ClickOptions, 'button' | 'clickCount'>
```
[the remark this answers is not in the document]
Comment (bvandrc) [reply src/spoof.ts:165]: fixed.

File: src/spoof.ts [hunk 2]
```diff
@@ -948,8 +946,20 @@ export class GhostCursor {
...
+          if (isElementHandle) {
+            const box = await destination.boundingBox()
+            if (box == null) {
+              throw new Error('no boundingBox')
+            }
+            return box
+          }
+
           return destination
+        }
```
[the remark this answers is not in the document]
Comment (bvandrc) [reply src/spoof.ts:972]: fixed

File: README.md [hunk 3]
```diff
@@ -164,11 +164,14 @@ Scrolls the element into view. If already in view, no scroll occurs.
...
-#### `scrollTo: (destination: Partial<Vector> | 'top' | 'bottom' | 'left' | 'right', options?: ScrollOptions) => Promise<void>`
+#### `scrollTo: (destination: Partial<Vector> | 'top' | 'bottom' | 'left' | 'right' | ElementHandle, options?: ScrollOptions) => Promise<void>`
 
 Scrolls to the specified destination point.
 
-- **destination:** An object with `x` and `y` coordinates representing the target position. For example, `{ x: 500, y: 300 }`. Can also be `"top"` or `"bottom"`.
+- **destination:** Can be any of:
+  - An object with `x` and `y` coordinates representing the target position. For example, `{ x: 500, y: 300 }`.
+  - `'top'` or `'bottom'` or `'left'` or `'right'`.
+  - An ElementHandle.
```
[the remark this answers is not in the document]
Comment (bvandrc) [reply README.md]: updated.

File: src/spoof.ts [hunk 4]
```diff
@@ -948,8 +952,18 @@ export class GhostCursor {
...
         case 'right':
           return { x: docWidth }
-        default:
+        default: {
+          if (isElementHandle(destination)) {
+            const box = await destination.boundingBox()
+            if (box == null) {
+              throw new Error('no boundingBox')
+            }
+            return box
```
Comment (Niek) [inline src/spoof.ts]: `boundingBox()` returns coordinates relative to the current viewport, but `scrollTo` later subtracts `scrollPositionTop`/`scrollPositionLeft` as though these were absolute document coordinates. This only reaches the correct target when the page starts at `(0, 0)`. I reproduced it with `scrollY = 1000`: targeting an element at document `y = 3000` ended at `scrollY = 2000`. Please convert the box coordinates to document coordinates (by adding the current scroll offsets), or use the box coordinates directly as deltas, and add a regression test that starts from a non-zero scroll position.

**Review by Niek — COMMENTED** [review]
One remaining issue found in the `ElementHandle` scroll path.

File: src/spoof.ts (hunk 4 above)
Comment (bvandrc) [reply src/spoof.ts]: done: https://github.com/Xetera/ghost-cursor/pull/207/commits/b355dbfdb86e884717409fd641f4218687a20132

also, added https://github.com/Xetera/ghost-cursor/pull/207/commits/662c069f53607027d208eb9b331fce49d61fdb40

**Comment by Niek** [comment]
Thanks a lot - merged now! Thanks for your contribution!
